where-to-travel/src/main/webapp/signup.js

```diff
@@ -77,17 +77,16 @@ $('#signUp-modal').on('hidden.bs.modal', function(){
  
 /** Signs up user using Firebase API **/
 function signUp() {
-  const promise = firebase.auth().createUserWithEmailAndPassword(emailSignUp.value, passwordSignUp.value);
-  promise.then(_ => {      
-    // Add user information to the real time database in Firebase
-    let ref = firebase.database().ref('users');
-    let data = {
-      name: nameSignUp.value,
-      email: emailSignUp.value,    
-      uID: firebase.auth().currentUser.uid
-    };
-    ref.push(data)
+  const promise = firebase().auth.createUserWithEmailAndPassword(emailSignUp.value, passwordSignUp.value);
+    promise.then(_ => {
+      let ref = firebase.database().ref('users/' + firebase.auth().currentUser.uid );
+      let data = {
+        name: nameSignUp.value,
+        email: emailSignUp.value,    
+        uID: firebase.auth().currentUser.uid
+      };
+    ref.set(data)
       .then(_ => $('#signUp-modal').modal('hide'))
-      .catch(e => {console.log(e.message); alert(e.message);});
-  }).catch(e => {console.log(e.message); alert(e.message);});
-}
\ No newline at end of file
+      .catch(_ => {console.log(_.message); alert(_.message);});
```

Comment:
for the last two catch statements, I think you would still want to keep e and use "e.message". I think you're only supposed to use "_" when you're not using anything from the response, similar to a function that is called with no passed in parameters.

Reply:
oh ok.  Thank you for the explanation that makes much more seance now!